python/deptry/dependency_getter/pep621/uv.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from deptry.dependency_getter.pep621.base import PEP621DependencyGetter
+from deptry.utils import load_pyproject_toml
+
+if TYPE_CHECKING:
+    from deptry.dependency import Dependency
+
+
+@dataclass
+class UvDependencyGetter(PEP621DependencyGetter):
+    """
+    Class to get dependencies that are specified according to PEP 621 from a `pyproject.toml` file for a project that
+    uses uv for its dependency management.
+    """
+
+    def _get_dev_dependencies(self, dev_dependencies_from_optional: list[Dependency]) -> list[Dependency]:
+        """
+        Retrieve dev dependencies from pyproject.toml, which in uv are specified as:
+
+        [tool.uv]
+        dev-dependencies = [
+            "pytest==8.3.2",
+            "pytest-cov==5.0.0",
+            "tox",
+        ]
```

Comment:
Maybe good to elaborate here that dev dependencies are extracted from there (`tool.uv`dev-dependencies` ) and also from the `dev_dependencies_from_optional` list. It took me some time to figure out what `dev_dependencies_from_optional` was and why it is needed here.

Reply:
Otherwise looks god to me and can be merged :)